test/test_ops.py

```diff
@@ -125,13 +131,162 @@ def test_inplace_gradgrad(self, device, dtype, op):
+                                                expected_grad, exception_during_backwards)
+
+    # Tests that the forward and backward passes of operations produce the
+    #   same values for the cross-product of op variants (function, method, inplace)
+    #   and runtimes (eager, traced, scripted).
+    # TODO WARNING: inplace x {traced, scripted} not currently tested
+    @ops(op_db)
+    def test_variant_consistency_jit(self, device, dtype, op):
+        samples = op.sample_inputs(device, dtype, requires_grad=True)
+        if len(samples) == 0:
+            self.skipTest("Skipped! No sample inputs!")
+
+        for sample in samples:
+
+            # Acquires variants to test
+            method = op.get_method()
+            inplace = op.get_inplace()
+            variants = (v for v in (method, inplace) if v is not None)
+
+            # Adds function variant to variant list
+            # TODO: inplace tests currently fail
+            # variants = (v for v in (op, method, inplace) if v is not None)
+            variants = (v for v in (op, method) if v is not None)
+
+            # Test traced and scripted consistency
+            for variant in variants:
+                # Create accessor for script function variant
+                if variant is op:
+                    name = op.name
+                    func_type = 'function'
+                elif variant is method:
+                    name = op.name
+                    func_type = 'method'
+                else:  # variant is inplace
+                    assert variant is inplace
+                    name = op.name + "_"
+                    func_type = 'inplace'
+
+                with disable_autodiff_subgraph_inlining():
+                    def fn(*inputs, **kwargs):
+                        attr = getattr(inputs[0], name)
+                        output = attr(*inputs[1:], **kwargs)
+                        return op.output_func(output)
+
+                    # Check scripted forward and grad
+                    script_fn = create_script_fn(self, name, func_type, op.output_func)
+                    check_against_reference(self, 
+                                            script_fn,
+                                            fn, 
+                                            (sample.input,) + sample.args, 
+                                            sample.kwargs, 
+                                            no_grad=(dtype not in floating_and_complex_types()))
+
+                    # Check traced forward and grad
+                    traced_fn = create_traced_fn(self, variant)
+                    check_against_reference(self, 
+                                            traced_fn,
+                                            fn, 
+                                            (sample.input,) + sample.args, 
+                                            sample.kwargs, 
+                                            no_grad=(dtype not in floating_and_complex_types()))
```

Comment:
the dtype functions are confusing but I think you need this one:

https://github.com/pytorch/pytorch/blob/febc76a5c69302c7d4b481632bcbd25e5d9b76ca/torch/testing/__init__.py#L293

and then you have to add half + bfloat16 manually, unfortunately